openassessment/xblock/openassessmentblock.py

```diff
@@ -364,12 +364,87 @@ def student_view(self, context=None):
+                      "url_grade_available_responses": "/grade_available_responses_view",
+                      "staff_assessment": false,
+                      "parent_id": "vertical_block_id",
+                      "url_base": "/student_view",
+                      "id": "openassessment_block_id"
+                     }, ...]
+                "ora_item_view_enabled" - enabled LMS API endpoint to serve XBlock view or not
+
+        Returns:
+            (Fragment): The HTML Fragment for this XBlock.
+        """
+        ora_items = context.get('ora_items', []) if context else []
+        ora_item_view_enabled = context.get('ora_item_view_enabled', False) if context else False
+        context_dict = {
+            "ora_items": json.dumps(ora_items),
+            "ora_item_view_enabled": 1 if ora_item_view_enabled else 0
+        }
+
+        template = get_template('openassessmentblock/instructor_dashboard/oa_listing.html')
+        context = Context(context_dict)
+        fragment = Fragment(template.render(context))
+
+        return self._update_and_return_fragment(fragment, 'CourseOpenResponsesListingBlock')
+
+    def grade_available_responses_view(self, context=None):
+        """Grade Available Responses view.
+
+        Auxiliary view which displays the staff grading area
+        (used in the Open Response Assessment tab in the Instructor Dashboard of LMS)
+
+        Args:
+            context: Not used for this view.
+
+        Returns:
+            (Fragment): The HTML Fragment for this XBlock.
+        """
+        student_item = self.get_student_item_dict()
+        staff_assessment_required = "staff-assessment" in self.assessment_steps
+
+        context_dict = {
+            "title": self.title,
+            'staff_assessment_required': staff_assessment_required
+        }
+
+        if staff_assessment_required:
+            context_dict.update(
+                self.get_staff_assessment_statistics_context(student_item["course_id"], student_item["item_id"])
+            )
+
+        template = get_template('openassessmentblock/instructor_dashboard/oa_grade_available_responses.html')
+        context = Context(context_dict)
+        fragment = Fragment(template.render(context))
+
+        return self._update_and_return_fragment(fragment, 'StaffAssessmentBlock')
+
+    def _update_and_return_fragment(self, fragment, initialize_js_func):
+        """
+        Auxiliary function to return updated fragment
+
+        """
         i18n_service = self.runtime.service(self, 'i18n')
         if hasattr(i18n_service, 'get_language_bidi') and i18n_service.get_language_bidi():
             css_url = "static/css/openassessment-rtl.css"
         else:
             css_url = "static/css/openassessment-ltr.css"
 
+        # backgrid.js v0.3.7
+        fragment.add_css_url(self.runtime.local_resource_url(self, "static/css/lib/backgrid.min.css"))
+        self.add_javascript_files(fragment, "static/js/lib/backgrid.min.js")
```

Comment:
New JavaScript libraries should be installed through NPM and loaded from `node_modules`. The non-minifed version should be used (relying on our app's asset pipeline to do the minification ourselves for production, so that we can debug the unminified version in development)

Documentation on this can be found in our developer documentation for [Developing User Interfaces](http://edx.readthedocs.io/projects/edx-developer-guide/en/latest/user_interface_development.html#adding-javascript-libraries) or our Confluence page for [adding a new JavaScript library](https://openedx.atlassian.net/wiki/display/FEDX/How+to+add+a+new+third+party+JavaScript+library+to+LMS+or+Studio).